mule-extensions-api/src/main/java/org/mule/runtime/extension/api/client/params/ComponentParameterizer.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.client.params;
+
+import org.mule.api.annotation.Experimental;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.util.DataSize;
+import org.mule.runtime.extension.api.client.ExtensionsClient;
+import org.mule.sdk.api.annotation.MinMuleVersion;
+
+/**
+ * Parameterizes a component executed or created through the {@link ExtensionsClient}
+ * <p>
+ * <b>NOTE:</b> Experimental feature. Backwards compatibility not guaranteed.
+ *
+ * @since 1.6.0
+ */
+@Experimental
+@MinMuleVersion("4.6.0")
+public interface ComponentParameterizer<T extends ComponentParameterizer> {
```

Comment:
Will the sdk-api reflect the same change if the source support is added?

Reply:
to be decided if that ever happens